cardano-node-chairman/src/Chairman/IO/Network/Socket.hs

```diff
@@ -17,42 +11,32 @@ module Chairman.IO.Network.Socket
 import           Control.Exception (IOException, handle)
 import           Control.Monad
 import           Data.Bool
-import           Data.Either
-import           Data.Eq
 import           Data.Function
 import           Data.Functor
 import           Data.Int
 import           Data.Maybe
-import           Foreign.C.Error (Errno (..), eCONNREFUSED)
 import           Network.Socket (Family (AF_INET), SockAddr (..), Socket, SocketType (Stream))
 import           System.IO (FilePath, IO)
 import           Text.Show (show)
 
-import qualified Data.List as L
-import qualified GHC.IO.Exception as IO
 import qualified Network.Socket as IO
 import qualified System.Directory as IO
 import qualified UnliftIO.Exception as IO
 
 -- | Check if a TCP port is open
 isPortOpen :: Int -> IO Bool
-isPortOpen port = handle (\(_ :: IOException) -> return False) $ do
+isPortOpen port = do
   socketAddressInfos <- IO.getAddrInfo Nothing (Just "127.0.0.1") (Just (show port))
   case socketAddressInfos of
-    socketAddressInfo:_ -> canConnect (IO.addrAddress socketAddressInfo)
+    socketAddressInfo:_ ->
+      handle (return . const @Bool @IOException False) $
+        canConnect (IO.addrAddress socketAddressInfo) $> True
```

Comment:
Moved the exception handling to around `canConnect`.

I'm intentionally excluding `getAddrInfo` because if that fails, I want the exception to propagate and be reported by tests.  This is because a failure at that point means it's impossible to test if the port is open and any information associated with the exception would be useful in diagnosing the problem.